### DISCUSSION TITLE
H/W support for energy monitoring

### DISCUSSION THREAD

**Discussion by Regen231** [microsoft/PlanetaryComputer#274] [General]
Hi Team,

Does the planetary hub allow us to monitor energy consumption for our codes using open-source python libraries such as pyRAPL and ECO2AI?
If I try to install them and run for the programs that I write on the jupyter notebook on the hub, would I get energy information?

Thanks and Regards,
Regen

**Comment by TomAugspurger** [comment]
Hi Regen. I've never used those tools, but you are able to install additional packages so presumably they'd work.

You might also be interested in the [Microsoft Cloud for Sustainability](https://learn.microsoft.com/en-us/training/modules/cloud-sustainability-overview/emissions-impact-dashboard?ns-enrollment-type=learningpath&ns-enrollment-id=learn-bizapps.get-started-sustainability), which captures some of these metrics. That would only be an option if you use your [own compute](https://planetarycomputer.microsoft.com/docs/concepts/computing/#use-your-own-compute) in your own Azure Subscription.

**Reply by Regen231** [reply]
I just tried installing both the libraries on the hub. While ECO2AI installation failed, pyRAPL was installed but threw exception such as PyRAPLCantRecordEnergyConsumption. So, I was wondering if the instance's hardware support for energy monitoring is enabled or not. How can we check that?
Thanks and Regards,
Regen

**Reply by TomAugspurger** [reply]
I'm not sure. In case it's relevant, these are running on AKS.
